gateway/ln-gateway/src/lib.rs

```diff
@@ -1493,9 +1516,11 @@ impl Gateway {
     }
 
     pub async fn handle_shutdown_msg(&self, task_group: TaskGroup) -> Result<()> {
-        if let GatewayState::Running { lightning_context } = self.get_state().await {
-            self.set_gateway_state(GatewayState::ShuttingDown { lightning_context })
-                .await;
+        // Take the write lock on the state so that no additional payments are processed
+        let mut state_guard = self.state.write().await;
+        if let GatewayState::Running { lightning_context } = state_guard.clone() {
```

Comment:
Can we do something similar here as I mentioned above?

Reply:
See above comment

Reply:
As discussed on the call, we need to clone here since we need the `LightningContext` in the `ShuttingDown` state.